src/helper/options.js

```diff
@@ -0,0 +1,93 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | '_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+/**
+ * Add options support on any class
+ * - getOption : fn(key)
+ * - getOptions : fn()
+ * - prepareOptions : fn()
+ * - setOptions : fn(opts)
+ *
+ * @warning must be used with decorators on class definition
+ */
+define([
+            'jquery',
+            'ju-shared/class'
+        ],
+        function(
+            $,
+            Class
+        ) {
+    'use strict';
+
+    var OptionsHelper = Class.extend({
+        init : function() {
+            // Define options
+            this._opts = {};
+
+            // Define options collector
+            this._initOptsCollector(true);
+        },
+        /**
+         * Include options in the opts collector
+         * Set default options and override parents options
+         * Keeps the flow from child to parent (no untraceable flows with before and after super)
+         * - Child options always have precedence over parent options
+         * - Parent options should never overwrite children options
+         * @warn ALWAYS call setOptions BEFORE _super
+         * @param {object} class level options to queue
+         */
+        setOptions : function() {
+            this._optsCollector = this._initOptsCollector();
+            // Optimized array prepend
```

Comment:
it's optimized vs using `concat`? because I think this method could be written something like:

`this._optsCollector = [].concat(arguments).concat(this._optsCollector)`

or am I wrong?